libs/ui/src/hooks/auth/use_dipped_smart_card_auth.ts

```diff
@@ -0,0 +1,44 @@
+import useInterval from 'use-interval';
+import { DippedSmartCardAuth } from '@votingworks/types';
+import { useEffect, useState } from 'react';
+
+const AUTH_STATUS_POLLING_INTERVAL = 100;
+
+export interface AuthApiClient {
+  getAuthStatus: () => Promise<DippedSmartCardAuth.AuthStatus>;
+  logOut: () => Promise<void>;
+}
+
+export function useDippedSmartCardAuth(
```

Comment:
Could this hook be replaced by a `useQuery`?

Reply:
And in general, are you planning to use `useMutation` for the auth methods?

Reply:
I honestly hadn't been planning to since I didn't see much gain from using react-query for this data. But for consistency sake and for machines where we don't have an `AppContext` available, using react-query might make sense after all

Will try it out but would also be a good thing to discuss tomorrow!

Reply:
Yes, agreed that there's not much use for a cache but that consistency would be a benefit. Unified error handling could also be nice. I was also under the impression that we would try to get rid of `AppContext` by the end of this

Reply:
Realizing I wasn't doing a great job of distinguishing the short term from the long term!

> I was also under the impression that we would try to get rid of `AppContext` by the end of this.

Definitely! I still see this as a long term goal. I was just thinking about the best way to get backend auth wired up and running in the short term. That said, I don't think it's that much work to use react-query.

I had been envisioning one possible alternative long term solution where we create a more targeted `AuthContext` for auth since it is conceptually different than the rest of the react-query data (e.g. isn't persisted in the backend, is polled so doesn't require explicit invalidation on mutation, etc.). But the point about unified error handling is a compelling one in favor of just using react-query.

I think one of the subconscious things driving my hesitancy is the complexity and indirection that react-query adds. (Worth it for most data we store in it, to be clear.)

tldr: Leaning toward using react-query but a sanity check is worthwhile as we broaden the scope of what we use react-query for

Reply:
Yeah, I've wondered about `AuthContext` as a potential direction as well. I agree with most of your thinking here. Let's discuss and get to the bottom of it

Reply:
Discussed in person and we're aligned on the benefits and pain points of react-query. And despite the pain points (the key one being the clunkiness of error-handling-friendly mutations), agree that we should use react-query consistently as a wrapper around the `apiClient` ✅